apps/ex_wire/test/ex_wire/handshake_test.exs

```diff
@@ -36,321 +53,296 @@ defmodule HandshakeTest do
       her_static_private_key = ExthCrypto.Test.private_key(:key_b)
 
       set_environment(my_static_private_key)
 
-      handshake =
-        %Handshake{remote_pub: her_static_public_key}
-        |> Handshake.initiate()
+      my_handshake =
+        Handshake.new(her_static_public_key)
+        |> Handshake.generate_auth()
 
       set_environment(her_static_private_key)
 
-      {:ok, auth_msg, encoded_ack_resp, her_secrets} =
-        Handshake.handle_auth(handshake.encoded_auth_msg)
+      {:ok, her_handshake, her_secrets} =
+        Handshake.new_response()
+        |> Handshake.handle_auth(my_handshake.encoded_auth_msg)
+
+      # Same auth message, except remote has added the initiator ephemeral public key
+      assert my_handshake.auth_msg == remove_ephemeral_public_key(her_handshake.auth_msg)
+      assert %ExWire.Framing.Secrets{} = her_secrets
 
       set_environment(my_static_private_key)
 
-      {:ok, my_secrets, _frame_rest} = Handshake.handle_ack(encoded_ack_resp, handshake)
+      {:ok, my_updated_handshake, my_secrets} =
+        Handshake.handle_ack(my_handshake, her_handshake.encoded_ack_resp)
 
-      assert %ExWire.Framing.Secrets{} = her_secrets
+      assert my_updated_handshake.ack_resp == her_handshake.ack_resp
       assert %ExWire.Framing.Secrets{} = my_secrets
-      assert %ExWire.Handshake.Struct.AuthMsgV4{} = auth_msg
     end
   end
 
-  test "handshake build and handle auth msg / ack resp via eip-8" do
-    my_static_public_key = ExthCrypto.Test.public_key(:key_a)
-    my_static_private_key = ExthCrypto.Test.private_key(:key_a)
-    her_static_public_key = ExthCrypto.Test.public_key(:key_b)
-    her_static_private_key = ExthCrypto.Test.private_key(:key_b)
-
-    {my_auth_msg, my_ephemeral_key_pair, _nonce} =
-      Handshake.build_auth_msg(
-        my_static_public_key,
-        my_static_private_key,
-        her_static_public_key
-      )
-
-    {:ok, encoded_auth_msg} =
-      my_auth_msg
-      |> Handshake.Struct.AuthMsgV4.serialize()
-      |> Handshake.EIP8.wrap_eip_8(her_static_public_key, my_ephemeral_key_pair)
-
-    {:ok, her_auth_msg, <<>>} = Handshake.read_auth_msg(encoded_auth_msg, her_static_private_key)
-
-    # Same auth message, except we've added the initiator ephemeral public key
-    assert her_auth_msg.initiator_ephemeral_public_key != nil
-    assert my_auth_msg == %{her_auth_msg | initiator_ephemeral_public_key: nil}
-
-    {her_ack_resp, her_ephemeral_key_pair, _her_nonce} = Handshake.build_ack_resp()
-
-    {:ok, encoded_ack_msg} =
-      her_ack_resp
-      |> Handshake.Struct.AckRespV4.serialize()
-      |> Handshake.EIP8.wrap_eip_8(my_static_public_key, her_ephemeral_key_pair)
-
-    {:ok, my_decoded_ack_resp, _ack_bin, <<>>} =
-      Handshake.read_ack_resp(encoded_ack_msg, my_static_private_key)
-
-    assert my_decoded_ack_resp == her_ack_resp
-  end
+  describe "EIP-8 test vectors" do
+    test "handshake auth plain" do
```

Comment:
The changes that follow in this test are all just indentation. I put them under the `describe "EIP-8 test vectors"` to more clearly outline test groups.